### ISSUE TITLE
Integration with `ripgrep` not working

### ISSUE THREAD

**Issue by lebensterben** [lotabout/skim#365]
I have `sk 0.9.1` and `rg 12.1.1` installed, both via `cargo-install`.
Both `sk` and `rg` are in my path, and they work well independently.
But I cannot integrate them with `sk --ansi -i -c 'rg --color=always --line-number "{}"'`.
Nevertheless, `sk` is working with `grep`. ( I don't have `ag` or `ack` so I cannot test these ).

**Comment by lotabout** [comment]
Please describe what you’ve done, what you expect and what actually happen. Thanks.

**Comment by lebensterben** [comment]
- After typing the command and press enter, the program `rg` starts with no query pattern and search result, which is what's expected.
- But even after I entered a pattern, there's no feedback at all.
- For example, suppose in a directory where some files contain the pattern `foo`,
    - I can `rg foo` or `grep foo`, which yields same results.
    - I can use `sk`'s integration with `grep`, that after pressing enter I type `foo`, and it well search for this pattern with `grep`, and the result is then filterred by `sk`
    - But I'm not able to integrate `sk` with `rg`, with the aforementioned method, as described in `README`. There's just NO feedback at all.

**Comment by lotabout** [comment]
Could you remove “—color=always” and try again?

A known bug would break ansi display, not sure if that’s the cause.

**Comment by lebensterben** [comment]
@lotabout 
I believe I found the bug and it's not related to `color`.
What I've noticed is:
- When I tried to add `--preview="bat --color=always {}"`, and it shows a *correct* error message that `bat` is not found. This results from how I specify my `PATH` environment variable that non-interactive shell won't see `~/.cargo/bin`. And it's perfectly fine if I use `--preview="$HOME/.cargo/bin/bat --color=always {}"` instead.
- Then I suspect that the command `sk --ansi -i -c 'rg --color=always --line-number "{}"'` is not working because `rg` is not in the `PATH`.
    - At first I suspect `sk` cannot correctly parse the result of `rg`, which has some small differences from `grep`'s. But even after I made `rg` output the same stuff as `grep`, `sk` still doesn't work with `rg`.
    - I then found it worked perfectly fine if I use something like `rg -l FOOBAR | sk -i` or `rg --files | sk -i`. This doesn't make any sense if `sk` really cannot work with `rg`.
    - So I tried to specify the full path to `rg` as `sk --ansi -i -c '$HOME/.cargo/bin/rg --color=always --line-number "{}"'`. Now it *works*.

So, to sum up, this is my conclusion now.
- `sk` works perfectly okay with `rg`.
- Due to the way how I set up my `PATH`, `rg` is not in the `PATH` when it's called from non-interactively. Though I'm calling `sk` in an interactive shell, `sk` calls `rg` "non-interactively". And it cannot see `rg`.
- The real problems is, if there is a `command-not-found` error when trying to execute `-c COMMAND`, `sk` should report this to the user.

**Comment by lebensterben** [comment]
Also, a fully working command where I can use both `rg` for searching and `bat` for preview, is

```shell
sk --ansi -i --preview="$HOME/.cargo/bin/bat --color=always $PWD/{}" -c '$HOME/.cargo/bin/rg --color=always -l "{}"'
```

For majority of users it's as simple as

```shell
sk --ansi -i --preview="bat --color=always $PWD/{}" -c 'rg --color=always -l "{}"'
```

**Comment by lebensterben** [comment]
```shell
function ski () {
  sk --ansi -i --preview="bat --color=always" -c 'rg --color=always -l "{}"'
}
zle -N ski
bindkey '^T' ski
```

**Comment by lotabout** [comment]
Thanks! I’ll dig more into the that.
